plutarch-orphanage/src/PlutusLedgerApi/V2/Orphans.hs

```diff
@@ -0,0 +1,1271 @@
+          (\ix -> Left . (ix,))
+          (Right . Left)
+          (Right . Right . Left)
+          (Right . Right . Right . Left)
+          (Right . Right . Right . Right)
+      outOf ::
+        Either
+          (Integer, [PlutusTx.BuiltinData])
+          ( Either
+              [(PlutusTx.BuiltinData, PlutusTx.BuiltinData)]
+              ( Either
+                  [PlutusTx.BuiltinData]
+                  ( Either Integer PlutusTx.BuiltinByteString
+                  )
+              )
+          ) ->
+        PlutusTx.BuiltinData
+      outOf = \case
+        Left (ix, dats) -> Builtins.mkConstr ix dats
+        Right (Left kvs) -> Builtins.mkMap kvs
+        Right (Right (Left ell)) -> Builtins.mkList ell
+        Right (Right (Right (Left i))) -> Builtins.mkI i
+        Right (Right (Right (Right bs))) -> Builtins.mkB bs
+
+-- | @since 1.0.0
+deriving via PlutusTx.BuiltinData instance Arbitrary PLA.Datum
+
+-- | @since 1.0.0
+deriving via PlutusTx.BuiltinData instance CoArbitrary PLA.Datum
+
+-- | @since 1.0.0
+instance Function PLA.Datum where
+  {-# INLINEABLE function #-}
+  function = functionMap coerce PLA.Datum
+
+-- Note: This works because TxId and DatumHash are the same hash type, with the
+-- same length.
+
+-- | @since 1.0.0
+deriving via PLA.TxId instance Arbitrary PLA.DatumHash
+
+-- | @since 1.0.0
+deriving via PLA.TxId instance CoArbitrary PLA.DatumHash
+
+-- | @since 1.0.0
+instance Function PLA.DatumHash where
+  {-# INLINEABLE function #-}
+  function = functionMap coerce PLA.DatumHash
+
+-- | @since 1.0.0
+deriving via PlutusTx.BuiltinData instance Arbitrary PLA.Redeemer
+
+-- | @since 1.0.0
+deriving via PlutusTx.BuiltinData instance CoArbitrary PLA.Redeemer
+
+-- | @since 1.0.0
+instance Function PLA.Redeemer where
+  {-# INLINEABLE function #-}
+  function = functionMap coerce PLA.Redeemer
+
+-- | @since 1.0.0
+instance Arbitrary PLA.TxOutRef where
+  {-# INLINEABLE arbitrary #-}
+  arbitrary = PLA.TxOutRef <$> arbitrary <*> (getNonNegative <$> arbitrary)
+  {-# INLINEABLE shrink #-}
+  shrink (PLA.TxOutRef txI ix) = do
+    -- TxId doesn't shrink, so we don't bother
+    NonNegative ix' <- shrink (NonNegative ix)
+    pure . PLA.TxOutRef txI $ ix'
+
+-- | @since 3.1.0
+instance CoArbitrary PLA.TxOutRef where
+  {-# INLINEABLE coarbitrary #-}
+  coarbitrary (PLA.TxOutRef txI ix) = coarbitrary txI . coarbitrary ix
+
+-- | @since 3.1.0
+instance Function PLA.TxOutRef where
+  {-# INLINEABLE function #-}
+  function = functionMap into outOf
+    where
+      into :: PLA.TxOutRef -> (PLA.TxId, Integer)
+      into (PLA.TxOutRef txi ix) = (txi, ix)
+      outOf :: (PLA.TxId, Integer) -> PLA.TxOutRef
+      outOf (txi, ix) = PLA.TxOutRef txi ix
+
+-- Needed because otherwise, deriving an instance for TxInfo is impossible.
+-- Arguably this is actually a bug on IOG's side, as the corresponding Cardano
+-- type _does_ have an Ord instance.
+
+-- | @since 1.0.0
+deriving stock instance Ord PLA.ScriptPurpose
```

Comment:
https://github.com/IntersectMBO/plutus/blob/7dbbd7fe4e4ffd3393e6c5644a14fb610d92d524/plutus-ledger-api/src/PlutusLedgerApi/V2/Contexts.hs#L83

I just want y'all to appreciate this. How do they even construct the redeemers without the Ord?

Reply:
A mystery for the ages.